reportgen/analysis_modules/base.py

```diff
@@ -5,7 +5,7 @@
 
 class BaseAnalysis(ABC):
     @abstractmethod
-    def preflight(self, api):
+    def preflight(self, api, project_id):
```

Comment:
Should we just implement the projectID check here and make this not an `@abstractmethod`?  In most use cases this will be the desired behavior; in corner cases user can choose not to invoke the base `preflight`.

Reply:
Or maybe this is just the behavior for some components?

Reply:
Good suggestion. It may introduce some confusion around the attributes though since currently we do not have an init function in the base class. I would postpone this change and keep it in mid until we have a better idea of the common preflight functionalities that can be moved to the base class.

Reply:
Sounds good.  Just keep in mind that repeated code becomes challenging to maintain.